sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/internal/SubscriberAssembler.java

```diff
@@ -67,42 +86,54 @@ private SubscriberServiceClient newSubscriberServiceClient() throws ApiException
     }
   }
 
+  private SubscriberServiceClient getSubscriberServiceClient() {
+    return SUB_CLIENTS.computeIfAbsent(
+        options.subscriptionPath(), path -> newSubscriberServiceClient());
+  }
+
+  private CursorClient newCursorClient() throws ApiException {
+    return CursorClient.create(
+        CursorClientSettings.newBuilder()
+            .setRegion(options.subscriptionPath().location().extractRegion())
+            .build());
+  }
+
+  private CursorClient getCursorClient() {
+    return CURSOR_CLIENTS.computeIfAbsent(options.subscriptionPath(), path -> newCursorClient());
+  }
+
+  SubscriberAssembler(SubscriberOptions options, Partition partition) {
```

Comment:
Nit: I prefer to place important methods such as the constructor first so the file is easier to read.